hudi-spark-datasource/hudi-spark-common/src/main/java/org/apache/hudi/DataSourceUtils.java

```diff
@@ -284,6 +284,9 @@ public static HiveSyncConfig buildHiveSyncConfig(TypedProperties props, String b
         props.getString(DataSourceWriteOptions.HIVE_PASS_OPT_KEY(), DataSourceWriteOptions.DEFAULT_HIVE_PASS_OPT_VAL());
     hiveSyncConfig.jdbcUrl =
         props.getString(DataSourceWriteOptions.HIVE_URL_OPT_KEY(), DataSourceWriteOptions.DEFAULT_HIVE_URL_OPT_VAL());
+    if (hiveSyncConfig.hiveMetaStoreUri != null) {
+      hiveSyncConfig.hiveMetaStoreUri = props.getString(DataSourceWriteOptions.HIVE_URL_OPT_KEY(), DataSourceWriteOptions.DEFAULT_HIVE_METASTORE_URI_OPT_VAL());
+    }
```

Comment:
What is the purpose of the decision `hiveSyncConfig.hiveMetaStoreUri != null` ?

Reply:
Because this is not a required option .

Reply:
Looks weird, in which case the `hiveSyncConfig.hiveMetaStoreUri` can be null and what is the priority between the `hiveSyncConfig` and the `props` ?

Reply:
(1) When synchronizing hudi data with local hive,` hiveSyncConfig.hiveMetaStoreUri` can be set to null.
(2) There is no priority distinction between `hiveSyncConfig` and `props`. Because there is no props attribute in `hiveSyncConfig`, only a single attribute is set.